RFC/src/RFC-0230_HTLC.md

```diff
@@ -1,6 +1,102 @@
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This document describes how to implement hashed time locked contracts as an extension to [mimblewimble](MimbleWimble).
+
+## Related RFCs
+## Description
+### Abstract
+Hashed Time Locked Contracts are time locked contracts that only pays out after a certain criteria has been met or refunds the originator if a certain period has expired. 
+
+#### Time Locked contracts
+In [mimblewimble](MimbleWimble) timelocked contracts can be accomplished by modifying the kernel of each transaction to include a block height. This allows a limit to how early in the blockchain lifetime the specific transaction can be included in a block. 
+
+This requires users constructing a transaction to:
+* MUST include a lockheight in the kernal of their transaction,
+* MUST include the lockheight in the signed message to stop tampering with the lockheight after the transaction was constructed.
+
+This add the following requirement to a miner:
+* MUST not add any transaction to the mined [block](block) that has not already exceeded its lockheight.
+
+This also adds the following requirement to a [base node](base node):
+* MUST reject any [block](block) that contains a kernel with a lockheight not already past the [currenthead](currenthead).
+
+### Mimblewimble N-of-N Multisig UTXO
+
+A normal Mimblewimble UTXO does not have a notion of being a [multisig] UTXO. The value of UTXO is hidden inside the commitment `C(v,r) = r·G + v·H` by virtue of being blinded. However, the blinding factor `r` can be composed of multiple blinding factors where `r = r1 + r2 + ... + rn`. The output commitment can then be constructed as `C(v,r) = r1·G + r2·G + ... + rn·G + v·H = (r1 + r2 + ... + rn)·G + v·H` where each participant keeps their private blinding factor hidden and only provides their public blinding factor. A multi-party aggregated signature scheme like Musig may be employed to sign such a transaction so that all parties' interests can be protected.
+
+The base layer is oblivious as to how the commitment and related signature were constructed. To open such commitments (in order to spend it) only the n-of-n blinding factor `r` is required. The parties that wants to open the commitment needs to collaborate n-of-n blinding factor `r`.
+
+#### Hashed Time Locked Contract
+Unlike Bitcoin where this can be accomplished with a single transaction, in [MimbleWimble](MimbleWimble) this is a multi-step process to construct a timelocked contract. 
+
+The steps are as follows:
+* The sender MUST pay all the funds into a n-of-n [multisig](multisig) [utxo](UTXO).  
```

Comment:
Before getting into this it would be useful to describe what the net result of this contract is, i.e.

Alice is paying N parties, but x happens if... and y happens if..